### DISCUSSION TITLE
Is "auto-render" file-call necessary?

### DISCUSSION THREAD

**Discussion by stapeleliz** [KaTeX/KaTeX#3651] [Q&A] [answered]
I put all my math coding inside <script> tags. In Firefox's console, I'm seeing the following message:

>  Uncaught Error: No element provided to render
>     s https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/contrib/auto-render.min.js:1

Since I don't do inline coding with delimiters, can I (should I) delete the call to the auto-render extension?

Thank you!

Sincerely,
Eliz. Stapel

**Comment by edemaine** [comment, marked as answer]
Are you using the [mathtex-script-type](https://github.com/KaTeX/KaTeX/tree/main/contrib/mathtex-script-type) extension?  If so, yes, you don't need to use auto-render and should probably remove its inclusion.

**Reply by stapeleliz** [reply]
I'm not using that extension. But since the auto-render script seems to be generating an "Uncaught Error", clearly I'm doing something wrong.

To clarify, *all* of my KaTeX calls within a given web page are formatted as:

` ...blah blah here's some math <div id="[name goes here]"></div><script>katex.render("[coding]", [name goes here]);</script> blah blah...`

If the auto-render script is looking for something like:

`  ...blah blah here's some math $$[coding]$$ blah blah...`

...and this sort of call is *not* being used anywhere on my site, is the "auto-render" script needed?

Thank you!

Sincerely,
Eliz. Stapel

**Reply by edemaine** [reply]
If you're calling `katex.render` directly, you don't need auto-render.

**Reply by stapeleliz** [reply]
Thank you!
